File: estagio/conting/conting/ident.c
/*-----------------------------ident.c---------------------------------*/

/* Funcao para identificar o ramo e definir os parametros de alteracao 
      [ Versao p/ Contingencia Simp./Multip. Pos/Mid Compensation ] 

---------------------------------------------------------------------- */
#include <stdlib.h>
#include <math.h>
#include <stdio.h>
#include "global.h"

/*----------------------Inicio da funcao identdc--------------*/
void identdc(int j,int k){
	ej1 = 0.e0;
	ek1 = 0.e0;  
  /*-------------<< Seja linha ou transformador >>------------*/
	if(itipo[j] <= 1 && itipo[k] <= 1){ 
	//cobre toda combinacao sem "slack"
		ej1 =  1.e0;
		ek1 = -1.e0; 
	}
	else{
		if(itipo[j] == 2){
			if(itipo[k] == 0 || itipo[k] == 1)
				ek1 = -1.e0;
		}
		else{
			if(itipo[j] == 0 || itipo[j] == 1)
				ej1 = 1.e0;
		}
	}
	//wjk1[j] = ej1;
	//wjk1[k] = ek1;
	/*------------------ Para Matriz Blinha  -----------------*/
	 return;
}
/*------------------------Fim da funcao identdc-----------------------*/

/*----------------------Inicio da funcao identca----------------------*/

/* Funcao para identificar o ramo e definir os parametros de alteracao
   [ Versao para Contingencia Multipla CA <Fluxo> Mid Compensation ] 
   [ e Tambem para Metodos 1P-1Q Iterativos                        ]  */


void identca(int j,int k,int i){
	ej2 = 0.e0;
	ek2 = 0.e0; 
	xjk[i] = 1.e40; 
	if(itipo[j] == 0){
		ej2 = 1.e0;
		xjk[i] = xkl[i];
	}
	if(itipo[k] == 0){
		ek2 = -1.e0;
		xjk[i] = xkl[i]; 
	}
	if(tap[i] != 0){
		if(itipo[j] == 0)
			ej2 = 1.e0; 
	}
	//wjk2[j] = ej2;
	//wjk2[k] = ek2;
	/*-------------------- Para Matriz B2linha----------------*/
	return;
}
/*------------------------Fim da funcao identca-----------------------*/
//Data de Atualizacao: 11/06/2003

File: estagio/conting/PIPELINE/ident.f
c23456781234567812345678123456781234567812345678123456781234567812345678
c   1       2       3       4       5       6       7       8       9
c ----------------------------------------------------------------------
c                     Refere-se a Matriz B1linha
c ----------------------------------------------------------------------
c Subrotina para identificar o ramo e definir os parametros de alteracao
c [ Versao p/ Contingencia Simp./Multip. Pos/Mid Compensation/DC Incr. ] 
c   
      subroutine identdc(jj,kk,ii)
c  
      include 'globvar.inc'
      include 'varcont.inc' 
     
      ej1 = 0.0
      ek1 = 0.0 
      
c23456789     << Seja linha ou transformador >>
      if( itipo(jj) .le. 1 .and. itipo(kk) .le. 1 ) then ! cobre toda combinacao sem "slack"
         ej1 = 1.
         ek1 = -1.
      else                     ! cobre as combinacoes que contenham "slack"
        if( itipo(jj) .eq. 2 ) then
          if( itipo(kk) .eq. 0 .or. itipo(kk) .eq. 1 ) then
            ek1 = -1.      ! ref. |-----| PQ  e  ref. |-----| PV               
          endif
        else
          if( itipo(jj) .eq. 0 .or. itipo(jj) .eq. 1 ) then
            ej1 = 1.       ! PQ |-----| ref.  e  PV |-----| ref. 
          endif
        endif       
      endif
                     
c Atribuicao wjk1() <-- ej1 & ek1

      wjk1(jj) = ej1
      wjk1(kk) = ek1
      
      return
      end    ! Fim da subrotina identdc  (p/ Matriz Blinha)


c ----------------------------------------------------------------------
c                     Refere-se a Matriz B2linhas
c ----------------------------------------------------------------------
c Subrotina para identificar o ramo e definir os parametros de alteracao
c   [ Versao para Contingencia Multipla CA <Fluxo> Mid Compensation ]
c  
      subroutine identca(jj,kk,ii)
c  
      include 'globvar.inc'
      include 'varcont.inc'  

      ej2 = 0.0
      ek2 = 0.0
      
c23456789
      xjk(ii) = 1.e+20
      if( itipo(jj) .eq. 0 ) then 
        ej2 = 1.
        xjk(ii) = xkl(ii)        ! verificar atraves de testes se e realmente necessario
      endif
      if( itipo(kk) .eq. 0 ) then 
        ek2 = -1.
        xjk(ii) = xkl(ii)        ! verificar atraves de testes se e realmente necessario
      endif
      if( tap(ii) .ne. 0. ) then
        if( itipo(jj) .eq. 0 ) ej2 = 1./tap(ii)    ! PQ |-----| ref., PV, PQ
      endif

                    
c Atribuicao  wjk2() <-- ej2 & ek2
      wjk2(jj) = ej2
      wjk2(kk) = ek2
     
      return
      end    ! Fim da subrotina identca  (p/ Matriz B2linhas)




